### PR TITLE
Update groupby

### LINKED ISSUE #20519: Groupby on multiindex with missing data in group keys raises IndexError
#### Code Sample, a copy-pastable example if possible

```python
foo= pd.DataFrame([['x', np.nan, 1]], columns = ['A', 'B', 'C']).set_index(['A', 'B'])
foo.groupby(level = ['A', 'B']).C.sum()
IndexError: cannot do a non-empty take from an empty axes.

# while this is expected, as a comparison
bar = pd.DataFrame([['x', np.nan, 1]], columns = ['A', 'B', 'C'])
bar.groupby(level = ['A', 'B']).C.sum()
Series([], Name: C, dtype: int64)
```
#### Problem description
It would be better if pandas can either return the same result as the 'bar' case, or raise another meaningful exception and document this case somewhere (e.g. [here](http://pandas.pydata.org/pandas-docs/stable/groupby.html#groupby-missing)))

#### Expected Output
Series([], Name: C, dtype: int64)
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Linux
OS-release: 3.10.0-229.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.1
nose: None
pip: None
setuptools: None
Cython: None
numpy: 1.10.4
scipy: None
statsmodels: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.4.2
pytz: 2016.4
blosc: None
bottleneck: 1.0.0
tables: None
numexpr: 2.4.6
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: 0.9.2
apiclient: 1.2
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
boto: 2.22.1
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by proost**
- [ ] closes #20519
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -1279,7 +1279,10 @@ def _get_grouper_for_level(self, mapper, level):
             # Remove unobserved levels from level_index
             level_index = level_index.take(uniques)
 
-        grouper = level_index.take(codes)
+        if not level_index.tolist():
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: this is non performant and obscure
what are you trying to do?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/multi.py (hunk 1 above)
Comment (proost) [reply pandas/core/indexes/multi.py]: if index has garbage values like nan or none, make it possible to groupby. So check level_index,then empty just pass level_index

Comment (jbrockmendel) [reply pandas/core/indexes/multi.py]: @proost is the goal to check if level_index is non-empty?  i.e. could `if not len(level_index):` work?

Comment (minhoryang) [reply pandas/core/indexes/multi.py]: @jreback I think @proost is trying to resolve #20519, and needs some help to get a clue. (He chose this issue at PyConKR19 Sprint, and got lost in codes.)

Comment (jbrockmendel) [reply pandas/core/indexes/multi.py]: I'm reasonably confident that `if not len(level_index):` is the way to do this here

Comment (proost) [reply pandas/core/indexes/multi.py]: @jbrockmendel  Thank you for your feedback. I change my code you mentioned.

Comment (jbrockmendel) [reply pandas/core/indexes/multi.py]: great.  can you rebase

**Pull request closed without merging**